FullStackMVCAPP/FullStackMVCAPP/DataContext/Repositories/CastleRepository.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+
+namespace FullStackMVCAPP.DataContext.Repositories
+{
+    public class CastleRepository
+    {
+        public IList<FullStackMVCAPP.Models.Castle> EntityList(GOTContext _GOTContext)
+        {
+            return _GOTContext.Castles.ToList();
+        }
+
+        public FullStackMVCAPP.Models.Castle GetEntityByID(GOTContext _GOTContext, int id)
+        {
+            return _GOTContext.Castles.Where(chr => chr.Id == id).First();
```

Comment:
`First()` here implies to me that this ID could exist more than once in the database however, I know this shouldn't be the case as the ID field is the primary key. Using `Select()` here without the `Where()` might be a better way of querying the ID.